### PR TITLE
Renamed all Uniswap references to Vexchange, refactored some functions and tests to accommodate our fee setting infra

### PR DESCRIPTION
**Pull request by xenide**
1. What's our plan of managing the V2Factory using the governance model? Are we going to follow Uniswap's model entirely, meaning, we will deploy VEX, GovernorAlpha, Timelock, FeeTo, FeeToSetter, TreasuryVester. Then Timelock will own FeeToSetter to control all the parameters, and GovernorAlpha will own Timelock. 
- Who will be the feeRecipient for feeTo? 
- Who will be the recipient for TreasuryVester?

2. If indeed we are following Uniswap's model entirely (i.e. renouncing ownership of V2Factory in the near future to the feeToSetter contract), we would probably have to enhance FeeToSetter with all the new functionality of the factory introduced by us, i.e. setRecovererForPair, setPlatformFeeForPair, setDefaultSwapFee etc. Otherwise there is no way for governance to adjust those parameters by FeeToSetter.

### CODE REVIEW

File: contracts/GovernorAlpha.sol [hunk 1]
```diff
@@ -302,6 +302,6 @@ interface TimelockInterface {
     function executeTransaction(address target, uint value, string calldata signature, bytes calldata data, uint eta) external payable returns (bytes memory);
 }
 
-interface UniInterface {
+interface VexInterface {
```
Comment (OliverNChalk) [inline contracts/GovernorAlpha.sol]: Do we want to move `interface` to the top? Having it at the bottom seems confusing lol. Same would go for `TimelockInterface`

File: contracts/GovernorAlpha.sol [hunk 2]
```diff
@@ -3,13 +3,13 @@ pragma experimental ABIEncoderV2;
...
-    string public constant name = "Uniswap Governor Alpha";
+    string public constant name = "Vexchange Governor Alpha";
 
     /// @notice The number of votes in support of a proposal required in order for a quorum to be reached and for a vote to succeed
-    function quorumVotes() public pure returns (uint) { return 40_000_000e18; } // 4% of Uni
+    function quorumVotes() public pure returns (uint) { return 40_000_000e18; } // 4% of vex
 
     /// @notice The number of votes required in order for a voter to become a proposer
-    function proposalThreshold() public pure returns (uint) { return 10_000_000e18; } // 1% of Uni
+    function proposalThreshold() public pure returns (uint) { return 10_000_000e18; } // 1% of vex
```
Comment (OliverNChalk) [inline contracts/GovernorAlpha.sol]: Capitalize `vex` for consistency across files?

File: contracts/TreasuryVester.sol [hunk 3]
```diff
@@ -47,16 +47,16 @@ contract TreasuryVester {
...
             amount = vestingAmount.mul(block.timestamp - lastUpdate).div(vestingEnd - vestingBegin);
             lastUpdate = block.timestamp;
         }
-        IUni(uni).transfer(recipient, amount);
+        IVex(vex).transfer(recipient, amount);
     }
 }
 
-interface IUni {
+interface IVex {
```
Comment (OliverNChalk) [inline contracts/TreasuryVester.sol]: Another floor interface. Up to you if we move all of them to top of file or just leave them to preserve a minimal diff

**Review by OliverNChalk — COMMENTED** [review]
Looks good, some minor comments. Also,do we want to delete UniswapV2Factory/Pair.json? Now that we have Vexchange replacements

**Comment by xenide** [comment]
> Looks good, some minor comments. Also,do we want to delete UniswapV2Factory/Pair.json? Now that we have Vexchange replacements

Yes, now that the remaining test cases don't need the Pair contract, we can delete the V2Pair json file

File: contracts/GovernorAlpha.sol (hunk 1 above)
Comment (xenide) [reply contracts/GovernorAlpha.sol]: okay fixed

File: contracts/TreasuryVester.sol (hunk 3 above)
Comment (xenide) [reply contracts/TreasuryVester.sol]: fixed

File: contracts/GovernorAlpha.sol (hunk 2 above)
Comment (xenide) [reply contracts/GovernorAlpha.sol]: fixed

**Comment by xenide** [comment]
@OliverNChalk actually the Uniswap build jsons have been deleted since a few commits ago (and were substituted by our vexchange ones). If it still exists locally, it might be your git client didn't delete it properly or something

**Comment by xenide** [comment]
@OliverNChalk the Uniswap json dependency was imported as a package. So I deleted this dependency from package.json too

**Review by OliverNChalk — APPROVED** [review]
I don't think you checked the package.json change in? Otherwise this looks ready to merge

File: package.json [hunk 4]
```diff
@@ -18,7 +18,6 @@
   "devDependencies": {
     "@types/chai": "^4.2.12",
     "@types/mocha": "^8.0.3",
-    "@uniswap/v2-core": "^1.0.1",
```
Comment (xenide) [inline package.json:21]: @OliverNChalk this is the change we're talking about

**Review by leodsoft — APPROVED** [review]
Yep, looks good. I confirmed the tests are all running & passing, and reviewed the test-case interacting with the Vex V2Factory. Will be good to see some scenarios on testnet next.

**Pull request merged**